Lina's Exercise Addition for Conditionals in R

linamnt opened the pull request.

jdblischak commented on 04-cond.Rmd:
````diff
@@ -211,6 +211,23 @@ In this case, "either" means "either or both", not "either one or the other but
 > plot_dist(dat[, 10], threshold = 10, use_boxplot = FALSE)  # day (column) 10 - create histogram
 > plot_dist(dat[1:5, 10], threshold = 10)                    # samples (rows) 1-5 on day (column) 10
 > ```
+>   + As a further challenge, fill in the blanks of the function hist_max() below to create a function that does the following:  
````
Since your challenge is unrelated to the previous challenge, please make yours a new challenge with its own title.

jdblischak commented on 04-cond.Rmd:
````diff
@@ -211,6 +211,23 @@ In this case, "either" means "either or both", not "either one or the other but
+>```
+>hist_max <- function(dat, spread) {
+>  max_patient_inflammation <- ______(dat, __, max)
+>  difference = ___(max_patient_inflammation) - ___(max_patient _inflammation)
````
Please use `<-` and not `=` for assignment.

jdblischak commented on 04-cond.Rmd:
````diff
@@ -211,6 +211,23 @@ In this case, "either" means "either or both", not "either one or the other but
+    2.Plot the max values into a simple histogram using the function hist() if and only if the spread (difference between largest and smallest) of the max values for all the patients is greater than your desired spread value, otherwise, return the values.  
+
+
+>```
````
Please put this in a code chunk and set `eval=FALSE`. This will give it the proper syntax highlighting when the lesson is rendered.

jdblischak commented:
Thanks for the PR, @linamnt. I've added some inline comments. In addition to addressing these, could you please also add the solution to the exercise? Please put it in a code chunk and set `include=FALSE` so that the answer is not displayed in the lesson.

linamnt commented:
Thanks for the feedback John! I think I've addressed your comments, please
let me know if I've missed anything.
Best,
Lina

On Mon, Jan 18, 2016 at 7:32 PM, John Blischak <notifications@github.com>
wrote:

> Thanks for the PR, @linamnt <https://github.com/linamnt>. I've added some
> inline comments. In addition to addressing these, could you please also add
> the solution to the exercise? Please put it in a code chunk and set
> include=FALSE so that the answer is not displayed in the lesson.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/swcarpentry/r-novice-inflammation/pull/157#issuecomment-172692083>
> .
>



-- 
Lina Tran
MSc. Candidate | Frankland Lab <http://www.franklandlab.com/>
Department of Physiology
University of Toronto

jdblischak commented on 04-cond.Rmd:
````diff
@@ -212,6 +212,38 @@ In this case, "either" means "either or both", not "either one or the other but
+> ## Challenge - Putting together data analysis, conditionals, and plotting   {.challenge}
+>   + For this challenge, fill in the blanks of the function hist_max() below to create a function that does the following:  
+  
+> ```{r, eval=FALSE}
````
Don't put the challenge description in a code block. Instead start each line with `>`.

jdblischak commented on 04-cond.Rmd:
```diff
@@ -212,6 +212,38 @@ In this case, "either" means "either or both", not "either one or the other but
+>  difference <- max(max_patient_inflammation) - min(max_patient_inflammation)
+>  if (difference > spread) {
+>      hist(max_patient_inflammation)
+>  } else {
```
I am a little confused here by the use case. Why would the function only return the data if the histogram is not made? Shouldn't it return it either way? For example, this makes more sense to me.

```
if (difference > spread) {
  hist(max_patient_inflammation)
}
return(max_patient_inflammation)
```

jdblischak commented on 04-cond.Rmd:
````diff
@@ -212,6 +212,38 @@ In this case, "either" means "either or both", not "either one or the other but
+> ``` 
+> hist_max <- function(dat, spread) {
+>  max_patient_inflammation <- ______(dat, __, max)
+>  difference <- ___(max_patient_inflammation) - ___(max_patient _inflammation)
````
Typo in variable name: `max_patient_inflammation`

jdblischak commented on 04-cond.Rmd:
````diff
@@ -212,6 +212,38 @@ In this case, "either" means "either or both", not "either one or the other but
+
+> ``` {r, include=FALSE}
+> hist_max <- function(dat, spread) {
+>  max_patient_inflammation <- apply(dat, 1, max)
````
Please pay attention to the indentation. It should always be 2 spaces. In your function, you sometimes indent by 1 space and other times by 4 spaces.

jdblischak commented:
Good work, @linamnt. You're almost there.

chendaniely commented on 04-cond.Rmd:
````diff
@@ -212,6 +212,36 @@ In this case, "either" means "either or both", not "either one or the other but
+> 2.Plot the max values into a simple histogram using the function hist() if and only if the spread (difference between largest and smallest) of the max values for all the patients is greater than your desired spread value, otherwise, return the values.  
+
+
+> ``` {r}
````
probably want the `eval=FALSE` knitr param here so this block does not render

chendaniely commented on 04-cond.Rmd:
```diff
@@ -212,6 +212,36 @@ In this case, "either" means "either or both", not "either one or the other but
 
+
+> ## Challenge - Putting together data analysis, conditionals, and plotting   {.challenge}
+>   + For this challenge, fill in the blanks of the function hist_max() below to create a function that does the following:  
```
surround hist_max() in backticks like such \`hist_max()\`, so it appears in code text: `hist_max()`

chendaniely commented on 04-cond.Rmd:
```diff
@@ -212,6 +212,36 @@ In this case, "either" means "either or both", not "either one or the other but
+> ## Challenge - Putting together data analysis, conditionals, and plotting   {.challenge}
+>   + For this challenge, fill in the blanks of the function hist_max() below to create a function that does the following:  
+  
+> 1.For each patient (row) in a given file, find the max inflammation value and store these in a variable called 'max_patient_inflammation'.  
```
backtick `max_patient_inflammation`

chendaniely commented on 04-cond.Rmd:
```diff
@@ -212,6 +212,36 @@ In this case, "either" means "either or both", not "either one or the other but
+>   + For this challenge, fill in the blanks of the function hist_max() below to create a function that does the following:  
+  
+> 1.For each patient (row) in a given file, find the max inflammation value and store these in a variable called 'max_patient_inflammation'.  
+> 2.Plot the max values into a simple histogram using the function hist() if and only if the spread (difference between largest and smallest) of the max values for all the patients is greater than your desired spread value, otherwise, return the values.  
```
`hist()`

chendaniely commented:
This is what the comment on line 223 is addressing.  The code block is executing your example code which is not correct `R` code.

![screenshot from 2016-05-31 22-17-25](https://cloud.githubusercontent.com/assets/5782147/15696418/1c818428-277e-11e6-9036-3a282e367fe3.png)

also @linamnt I like the idea of this challenge question.  It combines a lot of information into a single question (using `apply` with `min` and `max` to make a calculation to create a plot given a condition).
However, it might be out of scope for the people who are doing the lesson.  Might be a better fit as an intermediate or capstone challenge question, since the goal of this chapter is to introduce conditoinals.

Also, none of the other challenge questions use the fill in the blank formatting.
To be more consistent, I think an empty function with comments on all the individual components would be better.  This way learners aren't looking at partial bits of code.  It might also be confusing since the `_` are of different lengths.

chendaniely commented:
@linamnt I'm closing this since we are pushing for a V5.4 milestone, please feel free to re-submit the PR
